tests/framework/pvc.go

```diff
@@ -61,13 +61,38 @@ func (f *Framework) CreateAndPopulateSourcePVC(pvcName string, podName string, f
 	return sourcePvc
 }
 
+// CreateAndPopulateSourcePVCForBlockPV Creates and populates a PVC using the provided POD and command for block PV
+func (f *Framework) CreateAndPopulateSourcePVCForBlockPV(pvcName string, podName string, fillCommand string) *k8sv1.PersistentVolumeClaim {
```

Comment:
I did not get you. Your example here still shows another method (CreateAndPopulateSourcePVCForBlockPV), just with pvcDef parameter instead pvcName. so we remain with two similar methods.  There is also a call to utils.NewPodWithBlockPVC vs utils.NewPodWithPVC
If we want to have one method instead of two, we will have to create either the FS or Block pvc def, and either the FS or Block Pod, in advance, and then call CreateAndPopulateSourcePVCForPV with pvcDef and podDef as parameters.

Reply:
I copied the name wrong, I meant to take away the block part. Basically we end up with 1 function, and when you call it you pass in the definition you want.

Reply:
Is it right that to do so, I need to create the pvc def and pod def before we call this function? Meaning, changing all places we call this function in cloner_test.go and datavolume_test.go?

Reply:
Yes, that would be an improvement on the current state and we should always try to improve the existing code as well, if it is something we are touching.